File: Trees/BinaryTrees/postordertraversalIterative.py
class BinaryTree:
    def __init__(self,val):
        self.val = val 
        self.left = None
        self.right = None

def postordertraversalIterative(root):

    stack = []
    result = [] 
    
    stack.append(root)

    while stack :
        node = stack.pop()
        result.append(node.val)

        if node.left is not None:
            stack.append(node.left)

        if node.right is not None:
            stack.append(node.right)

    return result


node1 = BinaryTree(1)
node2 = BinaryTree(2)
node3 = BinaryTree(3)
node4 = BinaryTree(4)
node5 = BinaryTree(5)

node1.left = node2
node1.right= node3
node2.left = node4
node2.right= node5

result = postordertraversalIterative(node1)
while len(result) != 0:
    val = result.pop()
    print(val , end= " ")

File: Trees/BinaryTrees/postordertraversalIterative.java
import java.util.*;
class Node{
	
	int val ;
	Node left;
	Node right ;

	public Node(int val){
		this.val = val;
		this.left = null;
		this.right = null;
	}
}


class postordertraversalIterative{

	public static Stack<Integer> traversal(Node root){
		Stack<Integer> result = new Stack<>();
			
		Stack<Node> stack  = new Stack<>();
		stack.push(root);
		
		while( !stack.empty()){
			Node node = stack.pop();
			result.add(node.val);

			if (node.left != null){
				stack.add(node.left);
			}
			if (node.right != null){
				stack.add(node.right);
			}
		}
		return result;
	}

	public static void main(String args []){
		Node node1 = new Node(1);
		Node node2 = new Node(2);
		Node node3 = new Node(3);
		Node node4 = new Node(4);
		Node node5 = new Node(5);
		
		node1.left = node2;
		node1.right= node3;
		node2.left = node4;
		node2.right= node5;
	
		Stack<Integer> result = traversal(node1);
		while(!result.empty()){
			int val = result.pop();
			System.out.print(val);
		}
			
	}	
	}
